app/services/leader_updater.rb

```diff
@@ -0,0 +1,47 @@
+class LeaderUpdater
+  def initialize(group, updated_leader_ids)
+    @group = group
+    @updated_leader_ids = updated_leader_ids
+    @leader_ids_before_update = group.leader_ids
+  end
+
+  def update
+    remove_former_leaders
+    add_new_leaders
+
+    notify_leaders(former_leader_userids, 'remove_group_leader')
+    notify_leaders(new_leader_userids, 'add_group_leader')
+  end
+
+  private
+
+  def add_new_leaders
+    group_memberships_for(new_leader_userids)
+      .update_all(leader: true)
+  end
+
+  def remove_former_leaders
+    group_memberships_for(former_leader_userids)
+      .update_all(leader: false)
+  end
+
+  def notify_leaders(user_ids, pusher_type)
+    leaders_before_update = User.where(id: @leader_ids_before_update)
+    User.where(id: user_ids).each do |user|
+      GroupNotifier.new(@group, pusher_type, user)
```

Comment:
@julianguyen before this, user was being set to current_user, but I think that's not what we want? It's possible that I accidentally set it to current_user somewhere along the line. 

With the way it was  before, the messages were all saying `"#{whoever was current user} has been made a leader"`, but I think it should be `#{some member who now has leader = true} has been made a leader"`

The code above reflects the latter now, let me know if that's not what you want.